components/homme/src/share/cxx/mpi/BoundaryExchange.cpp

```diff
@@ -61,6 +61,18 @@ BoundaryExchange::BoundaryExchange()
   m_recv_pending = false;
 
   m_diagnostics_level = 0;
+
+#ifdef HOMMEXX_ENABLE_FWD_SENS
+  if constexpr (std::is_same_v<ScalarValue,FadType>) {
+    // We need a user-defined mpi datatype. Since SFad are
+    // contiguous in memory, we can use MPI_Type_contiguous
+    MPI_Type_contiguous(1+HOMMEXX_SFAD_SIZE, MPI_DOUBLE, &m_scalar_dtype);
```

Comment:
THis was the major change needed for MPI to work. With SFAD size 1, I think we were effectively broadcasting only half of the actual data...